### PR TITLE
list battles by status

### PR DESCRIPTION
**Pull request by DeborahAlmeida**
## Objective and Motivation
- List all battles related to user
- List by status: Settled and On going

### CODE REVIEW

File: backend/battle/views.py [hunk 1]
```diff
@@ -69,3 +69,9 @@ def get_queryset(self):
 class BattleDetail(DetailView):
     model = Battle
     template_name = "battle/battle_detail.html"
+
+    def get(self, request, *args, **kwargs):
+        self.object = self.get_object()  # pylint: disable=attribute-defined-outside-init
+        team = Team.objects.filter(battle=self.object, trainer=self.request.user)
+        context = self.get_context_data(object=self.object, team=team)
+        return self.render_to_response(context)
```
Comment (rsarai) [inline backend/battle/views.py]: Please update the context on the `get_context_data`

Comment (DeborahAlmeida) [reply backend/battle/views.py]: Okay, thanks!

File: backend/battle/views.py [hunk 2]
```diff
@@ -69,3 +69,20 @@ def get_queryset(self):
 class BattleDetail(DetailView):
     model = Battle
     template_name = "battle/battle_detail.html"
+
+    def get(self, request, *args, **kwargs):
+        self.object = self.get_object()  # pylint: disable=attribute-defined-outside-init
+        context = self.get_context_data(object=self.object)
+        return self.render_to_response(context)
```
Comment (fjsj) [inline backend/battle/views.py]: Why do you need to override this `def get`?

File: backend/battle/views.py [hunk 3]
```diff
@@ -69,3 +69,20 @@ def get_queryset(self):
...
+
+    def get_context_data(self, **kwargs):
+        context = {}
+        if self.object:
+            context['object'] = self.object
+            context_object_name = self.get_context_object_name(self.object)
+            if context_object_name:
+                context[context_object_name] = self.object
```
Comment (fjsj) [inline backend/battle/views.py]: I seems those lines belong to the `DetailView`. You can get this behavior by doing `context = super().get_context_data(**kwargs)`, thereby avoiding copying this code.

File: backend/battle/views.py [hunk 4]
```diff
@@ -69,3 +69,20 @@ def get_queryset(self):
...
+        team = Team.objects.filter(battle=self.object, trainer=self.request.user)
+        kwargs['team'] = team
+        context.update(kwargs)
+        return super().get_context_data(**context)
```
Comment (fjsj) [inline backend/battle/views.py]: Here you should just `return context`

File: backend/battle/views.py [hunk 5]
```diff
@@ -69,3 +69,20 @@ def get_queryset(self):
...
+    def get_context_data(self, **kwargs):
+        context = {}
+        if self.object:
+            context['object'] = self.object
+            context_object_name = self.get_context_object_name(self.object)
+            if context_object_name:
+                context[context_object_name] = self.object
+        team = Team.objects.filter(battle=self.object, trainer=self.request.user)
+        kwargs['team'] = team
+        context.update(kwargs)
```
Comment (fjsj) [inline backend/battle/views.py]: I think you can change this to `context['team'] = team`

**Review by fjsj — CHANGES_REQUESTED** [review]
Please check comments

File: backend/battle/views.py (hunk 5 above)
Comment (DeborahAlmeida) [reply backend/battle/views.py]: Thank you!

File: backend/battle/views.py (hunk 2 above)
Comment (DeborahAlmeida) [reply backend/battle/views.py]: It's really misleading, thank you :)

**Review by rsarai — APPROVED** [review]

**Pull request merged**